backend/src/v2/compiler/argocompiler/argo.go

```diff
@@ -326,3 +326,12 @@ var launcherResources = k8score.ResourceRequirements{
 const (
 	tmplEntrypoint = "entrypoint"
 )
+
+// Here is the collection of all special dummy images that the backend recognizes.
+// User need to avoid these image names for their self-defined components.
+// These values are in sync with the values in SDK to form a contract between BE and SDK.
+// TODO(lingqinggan): clarify these in documentation for KFP V2.
```

Comment:
When I wrote the comments, I was thinking about the scenario @gkcalat brought up, where the user uses an image name that's the same as our dummy image names. It feel necessary for the users to be informed of their image names choices.

I also thought it would be a good idea to have a single source of all the constants we are using here. However, it feels that neither package argo compiler nor package driver is a good place to be the place to hold these constants. By defining the costants in either places, it introduces an unnecessary dependency between the driver and the compiler. I think it's a better idea to define them in the `kubernetesplatform` proto, and all packages, both backend and sdk, all import from the single source.